text/2019-03-19-calculated-commit-ts.md

```diff
@@ -0,0 +1,191 @@
+# Calculated Commit TS
+
+## Summary
+
+This RFC proposes a feature of calculating `commit_ts` instead of allocating
+`commit_ts` from PD to reduce latency of committing.
+
+## Motivation
+
+TiKV uses Percolator's transaction algorithm. When committing a transaction,
+there are a few steps to do:
+
+1. Prewrite all keys with the transaction's `start_ts`
+2. Allocate the `commit_ts` from PD.
+3. Commit the transaction with `commit_ts`.
+
+The `commit_ts` allocated from PD is important in providing Snapshot Isolation,
+however allocating it from PD is not the only possible way. It's possible to
```

Comment:
When we use it to join two independent sentences, we need to use either use a period or semicolon for the first sentence. 
```suggestion
However, allocating it from PD is not the only possible way. It's possible to
```